### DISCUSSION TITLE
Limit Hashtag and Mentions for Posts

### DISCUSSION THREAD

**Discussion by ohbus** [scaleracademy/twitter-backend-java#15] [Q&A] [answered]
It would be great if we could mention the **maximum** number of mentions and hashtags that a **particular post** can have.

I am up to your suggestions what number would you want to have here? 🤔 

- [ ] Maximum number of mentions ❓ 
- [ ] Maximum number of hashtags ❓ 

> I am requesting this information so there is memory optimization due to the efficient use of Collections in Java.

**Comment by ohbus** [comment]
We know we have a character limit of 240 ✔️ 

So now we can make a constraint that any username and hashtags should be at least have 💭 5 characters, then we will be able to have a maximum mentions of around 40 for each. 🤔  Though it comes around 48, we need to consider the `@` before each mention and `#` for each hashtag so a round figure of around 40 sounds good maybe 💪🏻 

How does that sound ❓

**Reply by jeesou** [reply]
Hey @ohbus, this seems fine to me, usually this is what the limitations should be. Now the exact specifications are up for discussion.

**Comment by championswimmer** [comment, marked as answer]
There won't be any realistic impact on the **_use of Collections in Java_** as mentioned in the post, even if literally every post consist entirely of only hashtags and mentions.
